### PR TITLE
fix(cmake/modules): correct missing includes for libbpf

### PR DESCRIPTION
**Pull request by leogr**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

/area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap

> /area libpman

> /area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

Fix dependencies resolutions for `libbpf` when not all deps are bundled.

See https://github.com/falcosecurity/libs/pull/1583#pullrequestreview-1784265268

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmake/modules/libbpf.cmake [hunk 1]
```diff
@@ -25,6 +25,8 @@ elseif(NOT USE_BUNDLED_LIBBPF)
         message(FATAL_ERROR "Couldn't find system libbpf")
     endif()
 else()
+    include(zlib)
```
Comment (Andreagit97) [inline cmake/modules/libbpf.cmake:28]: they should be already included by `libscap` cmake before arriving here, `include(libbpf)` is done only after the other 2 includes :thinking:

Comment (leogr) [reply cmake/modules/libbpf.cmake:28]: Indeed, the issue is `libscap` itself.

A cmake config time, this :point_down: can happen before `include(libelf)` (defined somewhere else in the `libscap` cmake files.

https://github.com/falcosecurity/libs/blob/9d8c0a4babbdd01752d5d86ed6dd23194b339fc0/userspace/libscap/engine/modern_bpf/CMakeLists.txt#L24

The loading order depends on the actual options passed to make. 

In my case, with an older version of libs and the following command:
```
cmake -DUSE_BUNDLED_DEPS=Off -DUSE_BUNDLED_LIBELF=Off -DUSE_BUNDLED_LIBBPF=On -DUSE_BUNDLED_ZLIB=On -DBUILD_FALCO_MODERN_BPF=ON -DBUILD_BPF=ON -DMINIMAL_BUILD=On ..
``` 

I was able to trigger this path:

```
CMake Error at /usr/share/cmake-3.22/Modules/ExternalProject.cmake:3146 (get_property):
  get_property could not find TARGET zlib.  Perhaps it has not yet been
  created.
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/ExternalProject.cmake:3271 (_ep_get_file_deps)
  /usr/share/cmake-3.22/Modules/ExternalProject.cmake:3719 (_ep_add_configure_command)
  build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/cmake/modules/libbpf.cmake:21 (ExternalProject_Add)
  build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/userspace/libscap/engine/modern_bpf/CMakeLists.txt:8 (include)
```

Generally speaking, cmakefiles should be self-contained and not rely on other inclusions that may (or not) happen before. Resolving loading order should be a cmake responsibility, not us :) 


Does it make sense? :thinking:

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**